python-checks/src/main/java/org/sonar/python/checks/MissingNewlineAtEndOfFileCheck.java

```diff
@@ -38,26 +33,15 @@
     tags = Tags.CONVENTION
 )
 @SqaleConstantRemediation("1min")
-public class MissingNewlineAtEndOfFileCheck extends SquidCheck<Grammar> implements CharsetAwareVisitor {
+public class MissingNewlineAtEndOfFileCheck extends PythonCheck {
   public static final String CHECK_KEY = "S113";
   public static final String MESSAGE = "Add a new line at the end of this file \"%s\".";
-  private Charset charset;
-
-  @Override
-  public void setCharset(Charset charset) {
-    this.charset = charset;
-  }
 
   @Override
   public void visitFile(@Nullable AstNode astNode) {
-    String fileContent;
-    try {
-      fileContent = Files.toString(getContext().getFile(), charset);
-    } catch (IOException e) {
-      throw new IllegalStateException("Could not read " + getContext().getFile(), e);
-    }
+    String fileContent = getContext().pythonFile().content();
     if (fileContent.length() > 0 && !fileContent.endsWith("\n") && !fileContent.endsWith("\r")){
-      getContext().createFileViolation(this, String.format(MESSAGE, getContext().getFile().getName()));
+      addFileIssue(String.format(MESSAGE, getContext().pythonFile().file().getName()));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "file"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)